File: solutions/javascript/src/algorithm/stack/Exercise4.js
import DynamicStackADT from "../../structure/dynamic/DynamicStackADT.js";

function countStack(stack) {
  const aux = new DynamicStackADT();
  let count = 0;

  while (!stack.isEmpty()) {
    aux.add(stack.getElement());
    stack.remove();
    count++;
  }

  while (!aux.isEmpty()) {
    stack.add(aux.getElement());
    aux.remove();
  }

  return count;
}

// Ejemplo de uso
const stack = new DynamicStackADT();
stack.add(10);
stack.add(20);
stack.add(30);

console.log("Cantidad de elementos:", countStack(stack)); // → 3

// Mostrar contenido
while (!stack.isEmpty()) {
  console.log(stack.getElement());
  stack.remove();
}

File: solutions/python/algorithm/stack/Exercise4.py
from structure.dynamic.DynamicStackADT import DynamicStackADT

def count_stack(stack):
    aux = DynamicStackADT()
    count = 0

    while not stack.is_empty():
        aux.add(stack.getElement())
        stack.remove()
        count += 1

    while not aux.is_empty():
        stack.add(aux.getElement())
        aux.remove()

    return count

# Ejemplo de uso
stack = DynamicStackADT()
stack.add(10)
stack.add(20)
stack.add(30)

print("Cantidad de elementos:", count_stack(stack))  # → 3

# Mostrar el contenido (verificar que quedó igual)
while not stack.is_empty():
    print(stack.getElement())
    stack.remove()